### PR TITLE
Merge, min size, debug_raise()

### PR DESCRIPTION
**Pull request by MikhailKrichanov**

### CODE REVIEW

File: BaseTools/ImageTool/ElfScan.c [hunk 1]
```diff
@@ -546,13 +543,18 @@ CreateIntermediate (
 
       memcpy (Segments[SIndex].Data, (UINT8 *)mEhdr + Shdr->sh_offset, (size_t)Shdr->sh_size);
 
-      Segments[SIndex].ImageAddress = Pointer;
+      Segments[SIndex].ImageAddress = Shdr->sh_addr;
```
Comment (vit9696) [inline BaseTools/ImageTool/ElfScan.c:546]: Should not this be `Pointer`?

File: BaseTools/ImageTool/ElfScan.c [hunk 2]
```diff
@@ -546,13 +543,18 @@ CreateIntermediate (
...
       Segments[SIndex].ImageSize    = Segments[SIndex].DataSize;
       Segments[SIndex].Read         = true;
       Segments[SIndex].Write        = false;
       Segments[SIndex].Execute      = true;
       Segments[SIndex].Type         = ToolImageSectionTypeCode;
-      Pointer                      += Segments[SIndex].DataSize;
+
+      if (Pointer == 0) {
```
Comment (vit9696) [inline BaseTools/ImageTool/ElfScan.c:553]: Should not this be above `Segments[SIndex].ImageAddress` assignment?

File: BaseTools/ImageTool/ElfScan.c [hunk 3]
```diff
@@ -583,13 +585,18 @@ CreateIntermediate (
         memcpy (Segments[SIndex].Data, (UINT8 *)mEhdr + Shdr->sh_offset, (size_t)Shdr->sh_size);
       }
 
-      Segments[SIndex].ImageAddress = Pointer;
+      Segments[SIndex].ImageAddress = Shdr->sh_addr;
```
Comment (vit9696) [inline BaseTools/ImageTool/ElfScan.c:588]: Same here.

File: BaseTools/ImageTool/ElfScan.c (hunk 1 above)
Comment (MikhailKrichanov) [reply BaseTools/ImageTool/ElfScan.c:546]: No.

File: BaseTools/ImageTool/ElfScan.c (hunk 2 above)
Comment (MikhailKrichanov) [reply BaseTools/ImageTool/ElfScan.c:553]: No.

File: BaseTools/ImageTool/ElfScan.c (hunk 3 above)
Comment (MikhailKrichanov) [reply BaseTools/ImageTool/ElfScan.c:588]: Pointer is only needed here https://github.com/acidanthera/audk/blob/merge-min_size-assert/BaseTools/ImageTool/ElfScan.c#L620

**Pull request closed without merging**